sid/contacts.py

```diff
@@ -38,10 +38,11 @@ def calculate_contacts(contact_models, contact_policies, states, params, date):
             if policy_start <= date <= policy_end and cp["is_active"](states):
                 cont *= cp["multiplier"]
         if not model["is_recurrent"]:
-            cont = _sum_preserving_round(cont.to_numpy()).astype(DTYPE_N_CONTACTS)
+            cont = _sum_preserving_round(cont.to_numpy().astype(DTYPE_N_CONTACTS))
+            cont = cont.astype(DTYPE_N_CONTACTS)
         columns.append(cont)
 
-    contacts = np.column_stack(columns)
+    contacts = np.column_stack(columns).astype(float)
```

Comment:
Why first convert everything to `DTYPE_N_CONTACTS` and then to `float`?

Reply:
That doesn't make sense, you are right. But `column_stack' does not preserve the dtype of the elements passed to it. Now I just set the dtype to `DTYPE_N_CONTACTS` after `np.column_stack'.